x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -586,7 +587,13 @@ else if (target.foldable()) {
                     }
                 }
             }
-
+            // If we're only selecting literals, we have to fix the row count
```

Comment:
Maybe rephrase to something like`.. literals, we have to still execute the aggregation to create the correct grouping buckets, in order to return the appropriate number of rows`